bin/run_docker.py

```diff
@@ -0,0 +1,239 @@
+        input_dir: {"bind": "/input", "mode": "ro"},
+    }
+
+    volumes = {}
+    for vol in mounted_volumes.keys():
+        volumes[vol] = mounted_volumes[vol]
+
+    return volumes
+
+
+def run_docker(project: str, submission_id: str) -> None:
+    """
+    A function to run a Docker container with the specified image and handle any exceptions that may occur.
+
+    This function will run a Docker container using the image specified by the
+    ``submission_id`` argument, and will mount the input/ and output/ directories
+    in the current working directory to the corresponding locations in the
+    container. If the container runs successfully, the function will copy any
+    generated predictions file in the /output directory to Synapse. If the
+    container fails, the function will store the error message in a log file on
+    Synapse.
+
+    Args:
+        project: The Synapse project ID where the submission is located.
+        submission_id: The ID of the submission to run.
+
+    Returns:
+        None
+    """
+    # Get the Synapse authentication token from the environment variable
+    synapse_auth_token: str = os.environ["SYNAPSE_AUTH_TOKEN"]
+
+    # Communication with the Docker client
+    client = docker.from_env()
+
+    # Log into Synapse
+    syn = synapseclient.login(silent=True)
+
+    # Login to the Docker registry using SYNAPSE_AUTH_TOKEN
+    client.login(
+        username="foo",
+        password=synapse_auth_token,
+        registry="https://docker.synapse.org",
+    )
+
+    # Mount the input/ and output/ volumes that will exist in the submission container
+    volumes = mount_volumes()
+
+    # Get the Docker image ID from the submission
+    docker_image = get_submission_image(syn, submission_id)
+
+    # Run the docker image using the client:
+    # We use ``detach=False`` and ``stderr=True``
+    # to catch for and log possible errors in the logfile.
+    try:
+        container = client.containers.run(
+            docker_image,
+            detach=False,
+            volumes=volumes,
+            network_disabled=True,
+            mem_limit="6g",
+            stderr=True,
+        )
+
+        log_text = container
+
+        # Assuming the predictions file is generated in the /output directory
+        predictions_csv_path = os.path.join(os.getcwd(), "output", "predictions.csv")
+        predictions_zip_path = os.path.join(os.getcwd(), "output", "predictions.zip")
+
+        # Retrieve the output CSV or ZIP predictions file and store it on Synapse
+        if os.path.exists(predictions_csv_path):
+            store_predictions_file(project, predictions_csv_path, submission_id)
+        elif os.path.exists(predictions_zip_path):
+            store_predictions_file(project, predictions_zip_path, submission_id)
+
+    except Exception as e:
+        log_text = str(e).replace("\\n", "\n")
+        create_log_file(log_filename=f"docker.log", log_text=log_text)
```

Comment:
NIT these "docker.log" strings don't need to be f-strings. Also, you could create a constant which contains that string and reuse it in all of these places